### ISSUE TITLE
Add ariaLabelledBy prop to KSwitch's

### ISSUE THREAD

**Issue by AlexVelezLl** [learningequality/kolibri#12743]
## Overeview

We have several uses of KSwitch in our codebase: https://github.com/search?q=repo%3Alearningequality%2Fkolibri%20%3CKSwitch&type=code. And some of them doesnt have a `label` set. We should use the new `ariaLabelledBy` prop in KSwitch to point to the element that has the label corresponding to the switch component.

### Prerequisites
We should have https://github.com/learningequality/kolibri-design-system/issues/806 merged before continue with this issue.

## Acceptance criteria
- [ ] All KSwitch components should have a label, either set by the `ariaLabelledBy` or by the `label` prop

**Comment by RONAK-AI647** [comment]
Hi @AlexVelezLl , I would like to give a try to this issue , I will  be happy if you assign it to me ,I will post whatever the outcome is within few days.

**Comment by AlexVelezLl** [comment]
Hey @RONAK-AI647! Thanks for your interest in picking this issue up! Unfortunately this issue depends on https://github.com/learningequality/kolibri-design-system/issues/806 to be completed and released. So its blocked for now. But I can ping you when https://github.com/learningequality/kolibri-design-system/issues/806 is complete! Meanwhile you can refer to our list of unassigned ["help wanted"](https://github.com/search?q=org%3Alearningequality+is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+no%3Aassignee&type=issues) or ["good first issue"](https://github.com/search?q=org%3Alearningequality+is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+no%3Aassignee&type=issues) issues across all our repositories 🤗.

**Comment by apoorvapendse** [comment]
Is this issue closed because of 
https://github.com/learningequality/kolibri-design-system/pull/808
?

**Comment by AllanOXDi** [comment]
Hi @apoorvapendse , No! it's not close- I can assign it to you if you would like to take it.

**Comment by apoorvapendse** [comment]
Yeah sure : 👍

**Comment by AllanOXDi** [comment]
Assigned!

**Comment by AlexVelezLl** [comment]
Hi @apoorvapendse! Are you working in this issue? If so, there is no problem, there is no rush. Just checking in to know if we should unassign :).

**Comment by yashhash2** [comment]
Hi @AlexVelezLl , I would like to give a try to this issue , I will be happy if you assign it to me ,this issue hasnt resolved if nobody is working in this i would like to work.

**Comment by AlexVelezLl** [comment]
Hi @yashhash2! Lets give some more time to @apoorvapendse to report back in case they are working in this issue. But you’re welcome to find [a "help wanted" issue with no assignee](<https://github.com/search?q=org%3Alearningequality+is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+no%3Aassignee&type=issues>) :)

**Comment by apoorvapendse** [comment]
Hey @AlexVelezLl, thanks for being so patient and considerate.
I am caught up in something.
Please feel free to assign the issue to @yashhash2 
All the best @yashhash2 !
Thanks

**Comment by yashhash2** [comment]
Actually I am already assigned with two issues rn so it's better if it is assigned to someone else.

**Comment by RONAK-AI647** [comment]
@AlexVelezLl  assign it to me plzz... m ready to make my first contribution in kolibri .

> Hi [@AlexVelezLl](https://github.com/AlexVelezLl) , I would like to give a try to this issue , I will be happy if you assign it to me ,I will post whatever the outcome is within few days.

**Comment by AlexVelezLl** [comment]
Hi @RONAK-AI647! Super!!! Thank you! I Will assign this to you.

**Comment by RONAK-AI647** [comment]
@AlexVelezLl Raised a pr , please take a look .

**Comment by MisRob** [comment]
Hi @RONAK-AI647,  I wanted to mention that [Learning Equality will be closed from December 23 to January 5](https://github.com/learningequality/kolibri/discussions/12956).

**Comment by karankoder** [comment]
Hi @AlexVelezLl , I noticed that this issue is still open, and it seems @RONAK-AI647 is not actively working on it. Could you please assign this issue to me? I'd like to take it up

**Comment by marcellamaki** [comment]
Hi @karankoder - thank you for your interest in contributing to Kolibri! I've noticed that you've volunteered for a few issues where other contributors are assigned. Due to the new year, and Learning Equality being closed for the last 2 weeks, the work on issues and PRs have been slower than usual. Sometimes, issues are actually dropped and we do reassign them, but we would like to give people a little more time due to the end of year holidays for them to come back to their contributions. If they don't return to them, we will make these issues available again.

I know most of our "help wanted" issues on Kolibri are currently assigned to other community contributors, which may be frustrating. We will continue to open and tag issues for community contribution now that we have returned from our 2 week end of year office closure, and there will be more opportunities for your contributions in the coming weeks. :) When you do ask for assignment, please volunteer for just one unassigned issue at a time, as this will make the process much smoother for you and for our team. Thank you again for joining the LE open source community and happy new year!

**Comment by karankoder** [comment]
Hi @marcellamaki ,

Thank you for your response and the guidance! I completely understand and appreciate the process. I’ll focus on one unassigned issue at a time and keep an eye out for new opportunities to contribute. Wishing you and the team a Happy New Year! 😊

**Comment by RONAK-AI647** [comment]
@karankoder I have raised a PR karan and I am working on the issues on it .Please follow up the instructions by the maintainers.🥰

**Comment by karankoder** [comment]
Thanks for letting me know, @RONAK-AI647!

**Comment by AadarshM07** [comment]
Hey this issue looks inactive for a while , would it be okay if i work on it?

**Comment by RONAK-AI647** [comment]
Hey @AadarshM07 , I am a bit busy with my exams , rest I am working on the pr ...please follow maintainer's instructions and see another issue.

**Comment by gowtham243-cs50** [comment]
Hi @AlexVelezLl i fixed the issue

<img src="https://github.com/user-attachments/assets/682f054e-b769-4957-8ac0-693f278e10d9" width="300">
<img src="https://github.com/user-attachments/assets/9c30ea45-0d22-43d7-857a-e221fd658f46" width="300">
<img src="https://github.com/user-attachments/assets/ea2d12da-2d26-4a8b-8950-c1e37af47aac" width="300">

if the fix is acceptable can you please assign it to me

**Comment by Pariyul** [comment]
I'd love to take on this issue! Please assign it to me, and I'll share the outcome in a few days.

**Comment by KeehWz** [comment]
@AlexVelezLl I'd love to take on this issue and I am willing to spend most of my time in it. Please assign this issue to me.

**Comment by MisRob** [comment]
Hi @Pariyul and @KeehWz, welcome.

If you haven't yet seen the contributing guidelines, I'd like to invite you to have a look at CONTRIBUTING.md.

Only [unassigned 'help wanted' issues](https://github.com/search?q=org%3Alearningequality+is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+no%3Aassignee&type=issues) are open for contribution. There are no free 'help wanted' now so I recommend to refresh the list on weekly basis as we keep adding. Before then, we can't assign you any issues. Thanks for your interest and hopefully some time later we can collaborate.

**Comment by MisRob** [comment]
Closed by https://github.com/learningequality/kolibri/pull/12918
